src/System.CommandLine.Hosting/HostingExtensions.cs

```diff
@@ -107,7 +107,7 @@ public static IHostBuilder UseCommandHandler(this IHostBuilder builder, Type com
                 && invocation.ParseResult.CommandResult.Command is Command command
                 && command.GetType() == commandType)
             {
-                invocation.BindingContext.AddService(handlerType, c => c.GetService<IHost>().Services.GetService(handlerType));
+                invocation.BindingContext.AddService(handlerType, c => c.GetRequiredService<IHost>().Services.GetService(handlerType));
```

Comment:
Why this change? I would think the `IHost` is a required service here.

Reply:
`GetRequiredService` would return service and ensure the service is not null, `GetService` would return `null` when no service found, think we expect `IHost` exists, so maybe better to use `GetRequiredService` 
<https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.getrequiredservice?view=dotnet-plat-ext-6.0>

Reply:
Think it should not be registered when `IHost` or the handler is `null`, maybe I could add a null check for that